### PR TITLE
Tradheli: fix bug for auto missions using continue after land

### PR DESCRIPTION
**Pull request by bnsgeyer**
With the recent capability in Copter 4.1 for helicopters to be able to land and then takeoff during an auto mission, it was discovered that the rotor was not allowed to spool up completely before the takeoff was initiated by the autopilot resulting in the aircraft spinning until it regained tail rotor effectiveness.
**How the autopilot determines runup complete and setting ground idle**
The reason for the premature takeoff is due to how the autopilot determines when spool up is complete. The run up time (H_RUNUP_TIME) is what it uses during spool up to determine run up complete (ie rotor at flight capable RPM). The autopilot counts up from when the motor interlock switch is enabled. When it reaches runup time it declares runup complete and the autopilot raises the collective to take off. When the motor interlock is disabled, the autopilot counts down until it reaches the critical speed (H_RSC_CRITICAL) and at that point declares that the rotor is below critical and the autopilot allows disarming if in an auto mode like RTL or. LAND.
**What happens in an auto mission with continue after land**
What happens in the continue after land situation is the aircraft lands and waits for rotor speed below critical. Once that occurs, the autopilot initiates spool up. But only counts up from the critical speed instead of counting from zero. So it you had the runup time set to 10 seconds and the critical speed at 50%, the autopilot would only count 5 seconds to declare below critical rotor speed and then count up 5 seconds (instead of the full 10 seconds) to declare runup complete. For heli’s that use the governor of the ESC (H_RSC_MODE set to 2), this may not be enough time for the ESC to complete its spool up and thus the autopilot will try to take off before the ESC is ready.

**How it was fixed in this PR**
Ground Idle for heli's has always been set once the rotor speed went below the critical rotor speed determined based on the runup time.  The critical rotor speed is not the speed at which the ground idle should be set because in most cases, the rotor is still spinning 700 to 1000 RPM.  It would be more appropriate to make the autopilot wait the full length of the runup time before declaring ground idle.  This forces the autopilot to use the full runup time for the subsequent spool up and launch.  The code already protects the user from going into other non-manual throttle modes while the aircraft is spooling down or spooling up.  which should keep users from switching into auto or loiter and trying to takeoff before the rotor is ready for flight.  Thus the change in this PR was to force the autopilot to wait the length of the runup time before declaring ground idle.

**Testing**
This PR was tested in SITL with Realflight.  Want to test it in an actual heli before merging

### CODE REVIEW

File: libraries/AP_Motors/AP_MotorsHeli_RSC.cpp [hunk 1]
```diff
@@ -382,6 +382,12 @@ void AP_MotorsHeli_RSC::update_rotor_runup(float dt)
     if (_runup_complete && (get_rotor_speed() <= get_critical_speed())) {
         _runup_complete = false;
     }
+    // if rotor ramp and runup are both at zero, then spooldown has been completed
+    if (get_rotor_speed() <= 0.0f) {
```
Comment (rmackay9) [inline libraries/AP_Motors/AP_MotorsHeli_RSC.cpp:386]: you could of course shorten this to a single line but it's a style thing..

Comment (bnsgeyer) [reply libraries/AP_Motors/AP_MotorsHeli_RSC.cpp]: Done.  Thanks!

**Comment by bnsgeyer** [comment]
Tested this PR with my Synergy 626.  Works as expected.  Will merge once it passes all checks
